api/v1alpha4/types.go

```diff
@@ -58,6 +58,10 @@ type NetworkSpec struct {
 	// APIServerLB is the configuration for the control-plane load balancer.
 	// +optional
 	APIServerLB LoadBalancerSpec `json:"apiServerLB,omitempty"`
+
+	// LoadBalancerNodeOutboundIPs is the configuration for multiple frontend ips for node outbound load balancer.
+	// +optional
+	LoadBalancerNodeOutboundIPs *int32 `json:"load_balancer_node_outbound_ips,omitempty"`
```

Comment:
one of the other (sort of related) feature that has been requested that we might want to implement in the future is being able to disable the outbound LB all together https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1159

Not saying we should do it in this PR, but it would be good to think about how the UX might work for that and how this API can be modified later to include this, either by having a separate LB section for outbound, like `NodeOutboundLB.FrontendIPsCount`, or maybe reusing this field and allow setting it to 0 for no outbound IPs. WDYT?

Reply:
I'm inclined to having a separate section for outbound LB because:
1. Setting `LoadBalancerNodeOutboundIPs` to 0 will mean that the LB will have 0 public IPs, but there'll still be an outbound LB. Instead we can do a simple nil check to disable outbound LB, avoids any ambiguity.
2. We can have more control over the LB config like we do for api server LB.

Reply:
+1, a separate section makes sense to me as well

Reply:
Does it make sense to create the new LB field now so we don't have to change the API again later?

Reply:
yeah, make sense. I'll update the code.